### PR TITLE
Fix 2 problems with reverse_bits on all 3 circuit drawers

### PR DESCRIPTION
**Pull request by enavarro51**
### Details and comments

This PR fixes 2 problems with using `reverse_bits `with the circuit drawers.

Problem 1:
Conditional gates and measures have a double line that draws from the gate to the classical register or bit that it's assigned to. The function `_get_layered_instructions `in `qiskit.visualization.utils` will make sure that if while going through the dag, a gate would normally be placed below a conditional or measure, that gate, or the conditional or measure, will get moved to the next layer to the right so that the lines do not go through the gate. 

This works fine with `reverse_bits `off, but with `reverse_bits `on, it does not take into consideration that the gates above the conditional or measure will now be below, so this occurs.
```
       ┌───┐ ┌───┐  ┌─┐ 
 qr_2: ┤ H ├─┤ X ├──┤M├─
       ├───┤ └─╥─┘  └╥┘ 
 qr_1: ┤ H ├───╫─────╫──
       ├───┤ ┌─╫─┐ ┌─╫─┐
 qr_0: ┤ H ├─┤ X ├─┤ X ├
       └───┘ └───┘ └───┘
cr2_0: ═════════════════
            ┌──╨──┐  ║  
 cr_1: ═════╡     ╞══╩══
            │ = 2 │     
 cr_0: ═════╡     ╞═════
            └─────┘     
```
or
image

This PR now properly moves gates to the right to avoid this.

image

Problem 2:
For a conditional in the 'mpl' and 'latex' drawers, with `cregbundle `off, the bits in the classical registers are indicated to be on or off. When `reverse_bits `was set, the indicators did not reverse.
image
image
This now works correctly.
image

This PR fixes problem 2 in the 'mpl' drawer. The 'latex' drawer will be fixed in #5855.

### CODE REVIEW

File: test/python/visualization/test_circuit_text_drawer.py [hunk 1]
```diff
@@ -1320,6 +1320,70 @@ def test_text_conditional_1(self):
...
+        cr2 = ClassicalRegister(1, 'cr2')
+        qr = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(qr, cr, cr2)
+        circuit.h(0)
+        circuit.h(1)
+        circuit.h(2)
+        circuit.x(0)
+        circuit.x(0)
+        circuit.x(2).c_if(cr, 2)
+        circuit.measure(2, 1)
```
Comment (frankharkins) [inline test/python/visualization/test_circuit_text_drawer.py:1335]: Can you move the conditional gate to after the measurement please? Is a more likely scenario

**Review by frankharkins — CHANGES_REQUESTED** [review]
Looks good, except it looks like you've put conditionals before measurements in the tests due to #5958? If so, best to merge #5958 first, then this.

File: test/python/visualization/test_circuit_text_drawer.py (hunk 1 above)
Comment (enavarro51) [reply test/python/visualization/test_circuit_text_drawer.py:1347]: Done in https://github.com/Qiskit/qiskit-terra/pull/5938/commits/ed8beeba922bd94a31ec366507bfc60e1a4c2202

Comment (frankharkins) [reply test/python/visualization/test_circuit_text_drawer.py:1347]: @enavarro51 looks like you've done it in the mpl test but not the text test?

Comment (enavarro51) [reply test/python/visualization/test_circuit_text_drawer.py:1346]: Ah, so it is. Done in https://github.com/Qiskit/qiskit-terra/pull/5938/commits/6e4587854490ecb6461cb592e515864c3fccb4b6

Comment (frankharkins) [reply test/python/visualization/test_circuit_text_drawer.py:1346]: Perfect, thank you!

**Review by frankharkins — APPROVED** [review]
Looks good!

**Review by javabster — APPROVED** [review]
lgtm 🚀

**Review by 1ucian0 — APPROVED** [review]

**Pull request merged**